alacritty/src/display/mod.rs

```diff
@@ -758,6 +758,17 @@ impl Display {
             terminal.reset_damage();
         }
 
+        // Invalidate highlighted hints if grid has changed.
+        let mut filter_highlight = |highlight: &mut Option<HighlightedHint>| match highlight {
+            Some(hl) if !hl.has_valid_range(&mut terminal) => {
+                self.window.set_mouse_cursor(CursorIcon::Default);
+                *highlight = None;
+            },
+            _ => (),
+        };
+        filter_highlight(&mut self.vi_highlighted_hint);
+        filter_highlight(&mut self.highlighted_hint);
```

Comment:
Perhaps I'm thinking about this wrong because I frankly forget the distinction between `highlighted_hint` and `vi_highlighted_hint` exactly, but if there are two overlapping highlights, wouldn't we only want to change the cursor if both are invalid?

Reply:
You're both wrong in your conclusion and right in the assumption that this is a bug.

The vi highlight should have no impact on the mouse cursor at all. I'm not sure why I didn't think of that.